### ISSUE TITLE
[New Concept Exercise] : generators

### ISSUE THREAD

**Issue by BethanyG** [exercism/python#2293]
This issue describes how to implement the `generators` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)
- [Python Track concept exercise implementation guide](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md).

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the syntax and use of `generators` in Python. 

## Learning objectives

-  Understand what generators are and how/when to use them
-  Understand how generators relate to `loops` and `iterators`
-  Understand how to use the `yield` keyword
-  Understand the `__next__()` method
-  Create a generator

## Out of scope

- Memory and performance characteristics and optimizations
- `throw(type, value=None, traceback=None)`
- `close()`
- `generator expressions`
- `yield from`
- `generators` used as coroutines

## Concepts covered

-  `generators`
-  `yield`
-   `__next__()`
-  `iterators`

## Prerequisites

-  `conditionals`
-  `dicts`
-  `functions`
-  `higher-order-functions`
-  `lists`
-  `loops`
-  `iteration`
-  `iterators`
-  `sequences`

## Resources to refer to

-  [Generators (Python official docs)](https://docs.python.org/3/howto/functional.html#generators)
- [generator (Python official docs glossary)](https://docs.python.org/3/glossary.html#term-generator)
- [The yield statement (Python official docs)](https://docs.python.org/3/reference/simple_stmts.html#the-yield-statement)
- [Yield expressions (Python official docs)](https://docs.python.org/3/reference/expressions.html#yieldexpr)
- [Iterators(Python official docs)](https://docs.python.org/3/howto/functional.html?#iterators)
- [Generator-iterator methods (Python official docs)](https://docs.python.org/3/reference/expressions.html#generator-iterator-methods)
- [How to Use Generators and yield in Python (Real Python)](https://realpython.com/introduction-to-python-generators/)

### Hints

-  Referring to one or more of the resources linked above, or analogous resources from a trusted source.
- `Generators` section of the Python Docs Functional How to tutorial:  [Generators](https://docs.python.org/3/howto/functional.html#generators)

## Concept Description 

(_a variant of this can be used for the `v3/languages/python/concepts/<concept>/about.md` doc and this exercises  `introduction.md` doc._)

_**Concept Description Needs to Be Filled In Here/Written**_

_Some "extras" that we might want to include as notes in the concept description, or as links in `links.json`:_

-  Additional `Generator-iterator methods`, such as `generator.send()` and `generator.throw()`
- `generator expressions`
-  Asynchronous generator functions
- `generators` used as coroutines

## Representer

No changes required.

## Analyzer

No changes rquired.

## Implementing

The general Python track concept exercise implantation guide can be found [here](https://github.com/exercism/v3/blob/master/languages/python/reference/implementing-a-concept-exercise.md).

Tests should be written using `unittest.TestCase` and the test file named `generators_test.py`.

Code in the `.meta/example.py` file should **only use syntax & concepts introduced in this exercise or one of its prerequisites.** Please do not use comprehensions, generator expressions, or other syntax not previously covered.  Please also follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue, or contact one of the maintainers on our Slack channel.

**Comment by BethanyG** [comment]
@djjohnsongeek is going to take this on.  🎉 🌟

**Comment by J08K** [comment]
Heya, I am going to work on this.

## My idea
An airline needs your help with assigning seats for passengers, to do this they need **you** to make them a generator for that.

First, clarification:
A `seat` uses the following format `11F`, with each `1` being a number from 1 through 9, AKA the `rows`. And the `F` is the seat in each row, it uses `A through F`. It is important to note that a lot of airlines (we'll be using **all airlines** in the exercise) do not use row 13 because of the superstition surrounding it.

The exercise consists (prone to change) out of these functions/generators:
- `generate_seats`: Using the argument `number_of_seats` generate that amount of seats by counting up `1A, 1B, 1C, etc.` inside a generator.
- `assign_seats`: This generator needs to assign seats to a list of given `passengers`. It returns a list of tuples, with `(passenger_name, assigned_seat)`. These seats need to be generated by the previously mentioned function.

I think these should cover most of `generators`, but I am still stuck on how to implement `__next__()` and `next()` into the exercise itself.

**Comment by BethanyG** [comment]
No worries @djjohnsongeek!  We're happy you are back, and hopefully everything is going well for you.  
If you'd like, I think @J08K might be open to collaboration?   Just check in with him.  

But there are also other issues ([open issues](https://github.com/exercism/python/issues?q=is%3Aopen+is%3Aissue)) you can take on as well (if you'd like) -- just let us know!  😄

**Comment by BethanyG** [comment]
Closing this issue, since we've merged the `Plane Tickets` exercise.  Any further changes will be logged in a new issue.  Thanks everyone for your work!
